### PR TITLE
hyperparameter tuning and saving to S3

### PR DESCRIPTION
**Pull request by Anikkk**

### CODE REVIEW

File: Execute/execute_hyperparameters.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+def upload_data_to_s3(data, bucket_name, s3_key):
+    try:
+        s3.put_object(Bucket=bucket_name, Key=s3_key, Body=data)
+        print(f"Data uploaded successfully to S3: {s3_key}")
+    except Exception as e:
+        print(f"Error uploading data to S3: {e}")
+
+
+tuning('cotton')
```
Comment (VanshChiri) [inline Execute/execute_hyperparameters.py:89]: We need to make an API for tuning to be called. Will explain on call

File: application.py [hunk 2]
```diff
@@ -23,6 +24,16 @@ def forecast():
...
+@application.route('/hyperparameter', methods=['POST'])
+def hyperparameter():
+    try:
+        data = request.json
+        commodity_name = data.get('commodity_name')
+        tuning(commodity_name)
+        return jsonify({"message": "Forecasting started for " + commodity_name}), 202
+    except Exception as e:
+        return jsonify({"error": str(e)}), 500
+
```
Comment (kaivan-s) [inline application.py:36]: Have we verified calling this API?

File: Execute/execute.py [hunk 3]
```diff
@@ -183,4 +195,5 @@ def execute_model(model_func, raw_data, processed_data, forecast, hyperparameter
                                                                         hyperparameters)
     return actual_values, predictions, forecast_results, accuracy
 
-# forecast_pipeline('cotton')
\ No newline at end of file
+
+forecast_pipeline('cotton')
```
Comment (kaivan-s) [inline Execute/execute.py:199]: Remove this line or comment this line.

File: application.py (hunk 2 above)
Comment (Anikkk) [reply application.py:36]: image

image


Working !!

File: Execute/execute.py (hunk 3 above)
Comment (Anikkk) [reply Execute/execute.py]: removed

**Review by kaivan-s — APPROVED** [review]

**Pull request merged**